make hip-tests compileable with TheRock

cjatin opened the pull request:
## Motivation

Make changes through out hip-tests to make sure it compileable with TheRock

## Technical Details

- Upgrade Catch2
- Make language detection changes
- Packaging detection changes

## Test Plan

Existing tests should pass without any issues

## Test Result

*running tests*

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

cjatin committed:
SWDEV-1 upgrade to catch2v3

cjatin committed:
fixes

cjatin committed:
honor the offload arch by TheRock

cjatin committed:
Fix more bugs

Commit:
Enable hip-tests Rock build on Windows

- build and packaging zip file

Commit:
re-enable compiler and module tests Windows build

jayhawk-commits commented:
With the amount of files changed, I would feel more confident with SMEs from the hip runtime team reviewing, plus links to successful logs from both the legacy CI systems once the corresponding changes are made there plus logs from TheRock build system, when paired with ROCm/TheRock#2001

ScottTodd commented on projects/hip-tests/catch/CMakeLists.txt:
```diff
@@ -118,43 +93,28 @@ else()
+else()
+    # TheRock has already setup Catch2 for us
+    find_package(Catch2 REQUIRED)
 endif()
```
Copying a comment from https://github.com/ROCm/TheRock/pull/2001#discussion_r2496221622

I don't think a custom flag (especially not one referencing TheRock) should really be needed here. Other subprojects use `find_package` without such requirements. See https://github.com/ROCm/TheRock/blob/main/docs/development/dependencies.md for best practices. I think this general pattern should work in this case too: https://github.com/ROCm/rocm-libraries/blob/68b549eb4d2ad4a936526a60f79e0bdd91e7ad34/projects/hipblas/cmake/get-rocm-cmake.cmake#L6-L13

```cmake
find_package(PackageName ...)
if(NOT PackageName_FOUND)
  FetchContent_Declare(
    ...
  )
  FetchContent_MakeAvailable(...)
  find_pacakge(PackageName ... PATHS ...)
)
```

cjatin replied:
made some changes, have a look

ScottTodd replied:
Thanks, the changes in `projects/hip-tests/catch/CMakeLists.txt` look better to me now.

By the way, reviews are easier if you avoid rewriting git history and force pushing. Prefer to use merges instead of history rewriting rebases to sync your branch and push new commits in response to comments. I can't easily see what specifically changed between my prior comment and now:
* This force push compare shows unrelated changes:
<img width="876" height="51" alt="image" src="https://github.com/user-attachments/assets/09670eb6-4578-406c-9151-801a38713ee5" />

* "Changes since your last review" doesn't understand history rewrites: <img width="314" height="241" alt="image" src="https://github.com/user-attachments/assets/1fc5500b-4e69-4946-95f4-5d17b62b0fba" /> <img width="970" height="289" alt="image" src="https://github.com/user-attachments/assets/cee1294b-74f0-43ea-af2a-c9471f6bc35f" />

cjatin committed:
Fix some issues

cjatin committed:
rpm build fixes

cjatin committed:
Change catch script source location depending on if package is found or built

Commit:
Windows compute support for hip-tests catchv3

Commit:
rebase and fix tex build issues on linux

cjatin committed:
More fixes, change compiler to HIP for standalone builds, simplify lazy load

cjatin committed:
Disable grid sync for now

cjatin committed:
disable grid sync till we sort the issue out with compiler

cjatin committed:
use offload lists

agunashe committed:
skip installing of hipInfo.exe for linux builds

package_samples target currently just copies sample src files
support the same till the sample build is enabled for win and linux

agunashe committed:
skip gcc tests and remove module/coopKernel.code from custom target windows

agunashe committed:
skip rpm packaging if not found

agunashe committed:
copy catch2 libs to tests packages

ensures the test packages have the required libs for execution

agunashe committed:
pass --hip-path=<> for windows builds

disable two warnings
-Wno-ignored-attributes -Wno-microsoft-cast

agunashe committed:
call workaround script during hip-tests installation as well

without the workaround script. scripts have absolute paths

agunashe committed:
packaging hip-tests on windows Rock

cjatin committed:
remove rpath variables

cjatin committed:
add rpath search

agunashe committed:
fix json file generation using clang-cpp

using find_program to get path to clang-cpp

agunashe committed:
Windows: add failing tests to skip list

agunashe committed:
Windows: add the rock failing tests to skip list

agunashe committed:
set ROCM_PATH correctly to have default /opt/rocm

cjatin committed:
Honor ROCM_PATH to find rocm_agent_enum

agunashe committed:
add LD_LIBRARY_PATH relative to exe path

Since ctest/catch framework does not honor env varibles
Need to have this workaround for ctest to find the libs

agunashe committed:
Windows: Support hip-tests packaging on mlsejenkins
skip the last failing test on windows

agunashe committed:
Linux: skip failing, exception, timeout tests

agunashe committed:
Linux: Add all failing, exception, timeout tests

Run2 with rocm_path and overall timeout to 120 --> 300 mins

agunashe committed:
Change current test directory to exe path instead of script path

changing current test directory to exe path fixes many tests that use external targets
Add win mlsejenkins failures to skip list

agunashe committed:
reenable tests that are passing after the install exe path commit

agunashe committed:
skip building fp8_e8m0.cc in rock

agunashe committed:
skip the module tests failing on windows

ethtrinh committed:
Suppress deprecated-declaration warnings (#1817)

Co-authored-by: JeniferC99 <150404595+JeniferC99@users.noreply.github.com>

victzhan committed:
SWDEV-526773 - Modify LaunchDelayKernel to set a hard coded WallClock… (#1911)

* SWDEV-526773 - Modify LaunchDelayKernel to set a hard coded WallClock value when it's not avaliable

* Change hardcode clockrate in unit of KHz.

gandryey committed:
SWDEV-465041 - Enable direct dispatch under Linux by default. (#1934)

srihari-uttanur committed:
Update rocpd docs (#1276)

bkanango committed:
[SWDEV-556483] Fix runtime PM suspend causing test failures (#1931)

Added runtime PM detection and DRM ioctl-based device wake
to handle GPUs in BACO state. Modified tests to wake
suspended devices before reading sysfs files.

---------

Signed-off-by: Bindhiya Kanangot Balakrishnan <Bindhiya.KanangotBalakrishnan@amd.com>

anujshuk-amd committed:
[rocprof-sys] Fix test build failure on RHEL 10 (#1955)

## Motivation

To solve: SWDEV-566076 
FFmpeg versions >= 58.134 no longer expose read_seek and read_seek2 function pointers in AVInputFormat,
requiring alternative seek detection methods. This pull request updates the `VideoDemuxer` class to improve compatibility with newer versions of FFmpeg. The main change is how the code determines whether the input file is seekable, addressing differences in FFmpeg API versions.


## Technical Details

In `video_demuxer.h`, added a conditional check for `USE_AVCODEC_GREATER_THAN_58_134` to set `is_seekable_` to `true` for newer FFmpeg versions, since `read_seek` and `read_seek2` are no longer exposed in `AVFormatContext`. For older versions, the previous method of checking these fields remains in place. The conditional compilation
now assumes seek capability is available for newer FFmpeg versions.

victzhan committed:
SWDEV-567829 - populateFormatStringHashMap: relax printf hash collisi… (#1944)

* SWDEV-567829 - populateFormatStringHashMap: relax printf hash collision check for duplicate format strings

* function optimized by ai

itrowbri committed:
Updated rocprofv3.py to ignore old attach duration msec value (#1980)

marantic-amd committed:
[rocprof-sys] Process all information regarding agents and store them as extdata in rocpd database (#1880)

## Motivation

Resolved: SWDEV-566226

The current implementation of agents inside of rocprof-systems keeps just the minimal necessary set of information required for populating the `info_agent` table inside of rocpd database. There is a sufficient amount of data that is being left out from database, so this change should fix that and store the additional agent information as an `extdata` row inside of `info_agent` table.

## Technical Details

This PR introduces additional filed inside of `agent` structure inside which is representing the JSON formatted string of all the additional information we can acquire about particular agent. This data is processed and added during the initial fetching of agents, and afterwards pushed inside of the database.

---------

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

cadolphe-amd committed:
SWDEV-557412 - Incorporate proper chunk offset when remapping virtual memory (#1848)

* SWDEV-557412 - Incorporate proper offset when remapping virtual memory

* Fix condition to check if VMHeap allocation address matches a chunk address

* Move offset calculation outside if/else block

---------

Co-authored-by: JeniferC99 <150404595+JeniferC99@users.noreply.github.com>

gandryey committed:
SWDEV-567852 - Clean-up hip::init() (#1948)

kjayapra-amd committed:
SWDEV-559267 - Use CLPrint to DevLogPrintf with Log Level - detail debug. (#1160)

arsenm committed:
SWDEV-548892 - Stop using ocml isinf wrapper (#1854)

lttamd committed:
SWDEV-562708 - change default maximum SVM size to 256GB (#1731)

amilanov-amd committed:
SWDEV-503089 - Fix and enable disabled HIP tests from math group (#1319)

* SWDEV-503089 - Fix and enable disabled HIP tests from math group

* SWDEV-503089 - Move single precision reduced run to a common function

arsenm committed:
SWDEV-548892 - Stop using ockl steadyctr function (#1882)

Directly use the builtin

adam360x committed:
Implement PTL support (#1957)

* Implement PTL support

Signed-off-by: adapryor <Adam.pryor@amd.com>
(cherry picked from commit 45bc31292e7940a3b8fca044ef7df22047b95733)

Signed-off-by: Maisam Arif <Maisam.Arif@amd.com>

---------

Signed-off-by: adapryor <Adam.pryor@amd.com>
Signed-off-by: Maisam Arif <Maisam.Arif@amd.com>
Co-authored-by: Maisam Arif <Maisam.Arif@amd.com>

AidanBeltonS committed:
SWDEV-558080 - Add recommended granularity (#1176)

* Add recommended granularity

* Improve granularity testing

* Update based on feedback

manocharahul committed:
Fix and enable VMM tests on cuda (#1855)

* Fix and enable VMM tests on cuda

* Minor syntax fixes

---------

Co-authored-by: Rahul Manocha <rmanocha@amd.com>

jayhawk-commits commented on projects/hip-tests/catch/performance/event/CMakeLists.txt:
```diff
@@ -30,5 +30,4 @@ set(TEST_SRC
-                      TEST_TARGET_NAME build_tests
-                      COMPILE_OPTIONS -std=c++17)
\ No newline at end of file
+                      TEST_TARGET_NAME build_tests)
```
missing new-line at the end here

cjatin replied:
Done

jayhawk-commits commented on projects/hip-tests/catch/performance/memset/CMakeLists.txt:
```diff
@@ -35,5 +35,4 @@ set(TEST_SRC
-                      TEST_TARGET_NAME build_tests
-                      COMPILE_OPTIONS -std=c++17)
\ No newline at end of file
+                      TEST_TARGET_NAME build_tests)
```
missing new-line at the end

cjatin replied:
done

jayhawk-commits commented on projects/hip-tests/catch/unit/deviceLib/CMakeLists.txt:
```diff
@@ -85,7 +85,7 @@ set(AMD_TEST_SRC
     fp16_ops.cc
     fp8_host.cc
-    fp8_e8m0.cc
+    #fp8_e8m0.cc
```
Can a comment be added as to why this is now commented out?

cjatin replied:
Done, added a TODO

jayhawk-commits commented on projects/hip-tests/catch/unit/deviceLib/fp16_ops.cc:
```diff
@@ -73,7 +73,7 @@ void fp16_arith_cpu(const std::vector<float>& a, const std::vector<float>& b,
   constexpr size_t num_of_ops = 18;
   constexpr size_t iters = 100;
-  Catch::Generators::RandomFloatingGenerator<float> input1_gen(2.2f, 10.f);
+  Catch::Generators::RandomFloatingGenerator<float> input1_gen(2.2f, 10.f, /*seed*/ 0x1234);
```
Is the intention to always use the same seed?

cjatin replied:
At the moment yes, the old catch2 API did not accept any seed. The new one does.

jayhawk-commits replied:
Are there any concerns that there is a gap in testing if it's always the same seed?

cjatin replied:
I do not think so, but let me check the API of old catch2's interface and see how they handled seed.

jayhawk-commits commented on projects/hip-tests/catch/unit/gl_interop/CMakeLists.txt:
```diff
@@ -25,6 +25,5 @@ endif()
-                      TEST_TARGET_NAME build_tests
-                      COMPILE_OPTIONS -std=c++17)
+                      TEST_TARGET_NAME build_tests)
 target_link_libraries(GLInteropTest OpenGL::GL OpenGL::EGL GLUT::GLUT)
```
missing new-line here

cjatin replied:
done

jayhawk-commits commented on projects/hip-tests/catch/CMakeLists.txt:
```diff
@@ -382,3 +353,5 @@ if(DOXYGEN_FOUND)
     message(STATUS "Doxygen was not found. Building test plan documentation will not be available")
 endif()
+
+
```
One too many lines here

cjatin replied:
done

jayhawk-commits commented on projects/hip-tests/catch/cmake/relative_paths.py:
```diff
@@ -0,0 +1,109 @@
+inccmake_files = glob.glob(build_folder + "/**/*"+inc_cmake_pattern, recursive=True)
+make_test_files_portable(inccmake_files)
+
+
```
extra lines at the end here

cjatin replied:
done

jayhawk-commits commented on projects/hip-tests/catch/cmake/relative_paths.py:
```diff
@@ -0,0 +1,109 @@
+    #print(f"In relative_paths.py [1]: build folder: {build_folder}")
+    #print(f"In relative_paths.py [2]: install folder: {install_folder}")
+else:
+    print(f"Args not provided. Please provide arg1-build folder-catch_tests, arg2-install folder")
```
Perhaps providing an example of the expected args would be helpful here.

agunashe replied:
this script is a workaround script that handles some ctest/catch downsides and is only called within the hip-tests packaging build files. . I have an open item to completely get rid of this file and have a structural fix. 

Also, the example is in the comment. 
1st arg - build folder catch_tests
2nd arg - install folder the user wishes to install the tests.

jayhawk-commits commented on projects/hip-tests/catch/hipTestMain/hip_test_context.cc:
```diff
@@ -58,12 +58,27 @@ std::string TestContext::substringFound(std::vector<std::string> list, std::stri
-      "/opt/rocm/bin/rocm_agent_enumerator | awk '$0 != \"gfx000\"' | xargs | sed -e 's/ /;/g' | "
-      "tr -d '\n'";
+  constexpr std::string_view rocm_agent_enum = "rocm_agent_enumerator";
+  std::filesystem::path default_rocm_bin_path = "/opt/rocm";
```
Can we remove this default path value and still have it work?

cjatin replied:
No, we need to find `rocm_agent_enumerator`. We allow user to point to it with `ROCM_PATH`, if we do not find it, we need a fallback path.

jayhawk-commits replied:
OK. Note that this fallback probably should be removed for ROCm 8.0 onward.

cjatin replied:
Not sure what you mean here. hip-tests need to find rocm_agent_enum, users can point to it via setting environment variable `ROCM_PATH` which is what we do in TheRock enablement cmake change. But if user has not defined that variable fallback path is checking `/opt/rocm`. Are we changing the installation directory of `/opt/rocm` in ROCm 8.0?

kcossett-amd committed:
[rocprofiler-systems] Add support for ompt_callback_thread_begin (#1681)

* Add thread_begin callback

* Make OMPT callbacks that are instant have start_ts = end_ts

shadidashmiz committed:
SWDEV-567514: Remove default stream wait (#1977)

- when virtual map command is called

- can create deadlock

Signed-off-by: sdashmiz <shadi.dashmiz@amd.com>

iassiour committed:
Fix flaky test Unit_hipStreamAddCallback_StrmSyncTiming (#2022)

cjatin committed:
Review comments

agunashe committed:
Merge branch 'develop' into users/jatin/TheRock

agunashe committed:
Merge branch 'develop' into users/jatin/TheRock

agunashe committed:
skip the 3 failing tests to merge hip-tests rocm-systems PR

agunashe committed:
Merge branch 'develop' into users/jatin/TheRock

rahulc-gh reviewed (approved):
skipping windows CI gates based on - https://mlsejenkinsvm.amd.com/job/sandbox/job/hip-rt-psdb/9828/
